lib/shopping_cart.rb

```diff
@@ -0,0 +1,39 @@
+require_relative 'available_dishes'
+require_relative 'dish'
+
+class ShoppingCart
+
+  attr_accessor :cart_contents, :array_object
+  attr_reader :available_dishes
+
+  def initialize(available_dishes, cart_contents = [])
+    @cart_contents = cart_contents
+    @available_dishes = available_dishes
+    @array_object = nil
+  end
+
```

Comment:
I'm not sure if @available_dishes needs to be used here. If there is already an array of available dishes in the class AvailableDishes. Not sure if I've read it right though.